spec/lib/granite/represents_spec.rb

```diff
@@ -46,12 +55,40 @@ def model
 
   context 'when value was changed' do
     let(:attributes) { {field: 2} }
+    let(:new_value) { 3 }
 
-    it 'syncs before validation if value was changed' do
-      subject
-      expect_any_instance_of(Granite::Represents::Attribute).to receive(:sync)
-      subject.field = 3
-      subject.valid?
+    include_examples 'sync attribute value'
+  end
+
+  context 'when attrbute have default value' do
+    before do
+      stub_class(:model) do
+        include ActiveModel::Validations
+
+        attr_accessor :field
+      end
+
+      stub_class(:action, Granite::Action) do
+        allow_if { true }
+
+        represents :field, of: :model, default: 10
+
+        def model
+          @model ||= Model.new
+        end
+      end
+    end
+
+    context 'when new value equal default value' do
+      let(:new_value) { 10 }
+
+      include_examples 'sync attribute value'
+    end
+
+    context 'when new value not equal default value' do
+      let(:new_value) { 5 }
+
+      include_examples 'sync attribute value'
```

Comment:
please add a test that if a model has a value present BA field ignores `default` value

Reply:
Done.